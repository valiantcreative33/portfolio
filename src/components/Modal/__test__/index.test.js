import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();
const currentPhoto = {
    name: 'Park Bench',
    category: 'landscape',
    description: 'lorem ipsum dolor',
    index: 1
};

afterEach(cleanup);

describe('Modal Component', () => {
    it('renders', () => {
        render(<Modal currentPhoto={currentPhoto}/>);
    });

    it('matches snapshot DOM node structure', () => {
        const { asFragment } = render(<Modal 
            currentPhoto={currentPhoto}
            toggleModal={mockToggleModal}
        />);
        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Click Event', () => {
    it('calls onClose handler', () => {
        const { getByText } = render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
        />);
        fireEvent.click(getByText('Close this modal'));
        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    })
})