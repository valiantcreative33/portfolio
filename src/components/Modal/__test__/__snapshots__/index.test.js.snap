// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park Bench
      </h3>
      <img
        alt="current category"
      />
      <p>
        lorem ipsum dolor
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;